contributors/design-proposals/testing/test-frameworks-repo.md

```diff
@@ -0,0 +1,65 @@
+# New git repo: k8s.io/testframeworks
+
+Status: Pending
+
+Version: N/A
+
+Implementation Owner: TBD
+
+## Motivation: Why do we need a new repo?
+
+To avoid an vendoring cycle.
+
+Some folks in Sig-CLI have been writing an [integration testing
+framework](https://github.com/kubernetes/kubectl/tree/master/pkg/framework/test)
+which they would like to use to simplify some of their integration tests which
+live in `k8s.io/kubernetes`. The framework currently lives in `k8s.io/kubectl`.
+Since `k8s.io/kubectl` already vendors components _from_ `k8s.io/kubernetes`, it 
+cannot be vendored _into_ `k8s.io/kubernetes`. 
+
+It therefore seems like a good idea to create a new repo to contain testing
+frameworks. The intent is that this repo could be vendored anywhere.
+
+## Proposal
+
+We propose a new git repository named `k8s.io/testframeworks`. This would be
+backed by a github repository `github.com/kubernetes/testframeworks`.
+
+### What should live in it?
+
+The [integration testing
+framework](https://github.com/kubernetes/kubectl/tree/master/pkg/framework/test)
+should live in `k8s.io/testframeworks/integration`.
+
+There also [seems to be a
+desire](https://groups.google.com/a/kubernetes.io/d/msg/steering/LA9WiFnl6PI/os48-c3HCgAJ)
+to get `kubernetes/test/e2e/framework` out into a more reusable place. It could
+live in `k8s.io/testframeworks/e2e`
+
+### Other naming concerns
+
+We would like the name of the repo to be a reasonable [golang package
+name](https://blog.golang.org/package-names).  We should therefore probably
+avoid characters that are not lower case alphabetic characters. This precludes
+for example `k8s.io/test-frameworks`.
+
+### Who should own it?
```

Comment:
Also, what does ownership mean?

Reply:
In [this email](https://groups.google.com/a/kubernetes.io/d/msg/steering/LA9WiFnl6PI/os48-c3HCgAJ) when @spiffxp said "would naturally fall under the ownership of sig-testing", I presumed that "ownership" effectively meant something like "whose names go in the top level OWNERS file". I may have been wrong...

Reply:
I would expect the individuals proposing to work on this project to be the leads for it, and responsible for actually driving / implementing it.  This is not a thing I have the capacity to staff or drive.  So I'd like to see the "implementation owner TBD" part filled out.

But in the context of SIG's needing to own all things, in the same way that SIG Apps "owns" helm and charts, I think it makes sense for SIG Testing to own this.  I would like to hear from @timothysc if this aligns with his proposal to expand the charter of sig-testing but I'm personally +1 here

Reply:
I think the "individuals proposing to work on this project" are initially me, @hoegaarden and @apelisse. So far all the PRs on the [integration framework](https://github.com/kubernetes/kubectl/tree/master/pkg/framework/test) have been authored by me and @hoegaarden, and reviewed by @apelisse.

I can't speak with as much certainty for @apelisse , but I know that @hoegaarden and I are happy to be responsible for driving/implementing this framework so long as that makes sense for everyone else.

Reply:
I've added our (my, @apelisse, and @hoegaarden's) names to the proposal, and clarified my understanding of the difference between the "SIG's needing to own all things" context, and the "day to day looking after the code" context. Does that work for everyone?